crml/erc20-peg/src/lib.rs

```diff
@@ -233,10 +298,119 @@ decl_module! {
 				Erc20Meta::insert(address, (symbol, decimals));
 			}
 		}
+
+		#[weight = 1_000_000]
+		/// Sets the claim delay for a given AssetId
+		pub fn set_claim_delay(origin, asset_id: AssetId, min_balance: Balance, delay: T::BlockNumber) {
+			ensure_root(origin)?;
+			ClaimDelay::<T>::insert(asset_id, (min_balance, delay));
+			Self::deposit_event(<Event<T>>::ClaimDelaySet(asset_id, min_balance, delay));
+		}
 	}
 }
 
 impl<T: Config> Module<T> {
+	/// Process claims at a block after a delay
+	fn process_claims_at(now: T::BlockNumber, max_claims: u8) -> u8 {
+		let mut removed = 0_u8;
+		for (block, claim_id, pending_claim) in ClaimSchedule::<T>::iter()
+			.filter(|(block, _, _)| block <= &now)
+			.take(max_claims.into())
+		{
+			removed += 1;
+			match pending_claim {
+				PendingClaim::Deposit((deposit_claim, tx_hash)) => {
+					Self::process_deposit_claim(deposit_claim, tx_hash);
+				}
+				PendingClaim::Withdrawal(withdrawal_message) => {
+					// At this stage it is assumed that a mapping between erc20 to asset id exists for this token
+					let asset_id = Self::erc20_to_asset(withdrawal_message.token_address);
+					if asset_id.is_some() {
+						Self::process_withdrawal(withdrawal_message, asset_id.unwrap());
+					} else {
+						log::error!(
+							"📌 ERC20 withdrawal claim failed unexpectedly: {:?}",
+							withdrawal_message
+						);
+					}
+				}
+			}
+			ClaimSchedule::<T>::remove(block, claim_id);
+		}
+		removed
+	}
+
+	fn process_deposit_claim(claim: Erc20DepositEvent, tx_hash: H256) {
+		let event_claim_id = T::EthBridge::submit_event_claim(
+			&Self::contract_address().into(),
+			&T::DepositEventSignature::get().into(),
+			&tx_hash,
+			&EthAbiCodec::encode(&claim),
+		);
+		let beneficiary: T::AccountId = T::AccountId::decode(&mut &claim.beneficiary.0[..]).unwrap();
+		match event_claim_id {
+			Ok(claim_id) => Self::deposit_event(<Event<T>>::Erc20Claim(claim_id, beneficiary)),
+			Err(_) => {
+				// There was an error submitting an event claim. Could be that the bridge is down
+				// In this case, delay the deposit claim by 120 blocks
+				let claim_id = NextClaimId::get();
+				if !claim_id.checked_add(One::one()).is_some() {
+					log::error!("📌 No available claim ids");
```

Comment:
now that I think about it these could be made into events which is better for visibility